Bug 2001413: add/delete nic and disk to template

gouyang opened the pull request:
note for review:
1. rename tests/disks to tests/disk, tests/vm-templates to tests/template

gouyang commented:
/retest

gouyang commented:
/bugzilla refresh

gouyang commented:
/retest

gouyang commented:
/retest

gouyang commented:
/retest

gouyang commented:
/retest

pcbailey commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/create-template-from-wizard.spec.ts:
```diff
@@ -59,7 +59,7 @@ const pxeTemplate: VirtualMachineData = {
 };
 
-describe('Test VM creation', () => {
+describe('Test Template creation', () => {
```
nitpick: `Template` shouldn't be capitalized here.

gouyang replied:
NP

pcbailey commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:
```diff
@@ -0,0 +1,61 @@
+  size: '1',
+};
+
+describe('Test Template resource', () => {
```
nitpick: `Template` shouldn't be capitalized here.

gouyang replied:
NP

pcbailey commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:
```diff
@@ -0,0 +1,61 @@
+    deleteRow(disk1.name);
+  });
+
+  it('ID(CNV-7233) Delete user created template', () => {
```
nitpick: 'user created' => 'user-created'

gouyang replied:
NP

pcbailey commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:
```diff
@@ -0,0 +1,61 @@
+    });
+  });
+
+  it('ID(CNV-1850) Add/Delete a NIC to a VM template', () => {
```
nitpick: 'to' => 'to/from'

gouyang replied:
NP

pcbailey commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:
```diff
@@ -0,0 +1,61 @@
+    deleteRow(nic1.name);
+  });
+
+  it('ID(CNV-1849) Add/Delete a disk to a VM template', () => {
```
nitpick: 'to' => 'to/from'

gouyang replied:
NP

pcbailey reviewed (changes requested):
Just a few nitpicks. I would have went ahead and approved it and say we could fix them in a followup, but it needs a rebase to fix the `errorOccurred` change from https://github.com/openshift/console/pull/10017 as well.

gouyang commented:
/retest

gouyang commented:
@yaacov @pcbailey PTAL, thanks!
Note: I reverted the fix of flaky for tests/smoke/actions, added a clear wait `cy.wait(5000);` to `vm.start()` and `vm.restart()`.

pcbailey commented:
/lgtm

gouyang commented:
/retest

gouyang commented:
/retest

pcbailey commented:
/lgtm

pcbailey commented:
/lgtm
